pandas/tests/test_nanops.py

```diff
@@ -583,6 +583,103 @@ def test_nancorr_spearman(self):
         self.check_nancorr_nancov_1d(nanops.nancorr, targ0, targ1,
                                      method='spearman')
 
+    def test_nancorr_bicor(self):
+        a = self.arr_float_2d
+        b = self.arr_float1_2d
+        a_median = np.median(a)
+        b_median = np.median(b)
+
+        # Median absolute deviation
+        a_mad = np.median(np.abs(a - a_median))
+        b_mad = np.median(np.abs(b - b_median))
+
+        u = (a - a_median) / (9 * a_mad)
+        v = (b - b_median) / (9 * b_mad)
+
+        w_a = np.square(1 - np.square(u)) * ((1 - np.abs(u)) > 0)
+        w_b = np.square(1 - np.square(v)) * ((1 - np.abs(v)) > 0)
+
+        a_item = (a - a_median) * w_a
+        b_item = (b - b_median) * w_b
+
+        targ0 = (a_item * b_item).sum() / (
+            np.sqrt(np.square(a_item).sum()) *
+            np.sqrt(np.square(b_item).sum()))
+
+        a = self.arr_float_2d.flat
+        b = self.arr_float1_2d.flat
+        a_median = np.median(a)
+        b_median = np.median(b)
+
+        # Median absolute deviation
```

Comment:
You are simply copy and pasting the code from your implementation three times here! Those sort of tests provide very little value. The other unit tests is in this module can get away with less because they are relying on the well-tested scipy routines.

I would like to see tests with actual, fixed numbers that verify that bicor gives the right results. Typically, these would be in several edges cases as well as a few realistic small cases where you calculate the correct result by hand (or with another program). The edge cases should include things like NaNs and zeros.

You might also make some larger test cases with random numbers and verify that some invariants are satisfied (e.g., that bicor correlation is bigger than pearson correlation under particular circumstances, as was documented in some reference paper).

Copy other people's test cases is also usually a good thing, but please be careful before copying from R packages (or even looking closely at their implementations) because they are often only GPL licensed, which is incompatible with the pandas license.